addons/hr_holidays/security/hr_holidays_security.xml

```diff
@@ -138,11 +138,14 @@
     <record id="hr_leave_allocation_rule_multicompany" model="ir.rule">
         <field name="name">Time Off: multi company global rule</field>
         <field name="model_id" ref="model_hr_leave_allocation"/>
+        <!-- holiday_status_id is required, but may not be set when creating a new record -->
         <field name="domain_force">[
             '|',
                 ('employee_id', '=', False),
                 ('employee_id.company_id', 'in', company_ids),
-            ('holiday_status_id.company_id', 'in', company_ids + [False])
+            '|',
+                ('holiday_status_id', '=', False),
```

Comment:
This looks fishy, the field is required, so the NOT NULL constraint would be triggers is not set at creation. 

Why do you need to update the ir.rule ?

Reply:
Because the access rules are also checked when creating a transient record (when you create a new instance). In that case, the compute of holiday_status_id may assign a False when the default value is not set.

Reply:
Access rules are not checked on new records (during onchange).
The first time you check them is right after they are created (at the end of `create()`).

Reply:
I will make a better fix.